dataproc/single_job_workflow.py

```diff
@@ -0,0 +1,215 @@
+                 cluster_name):
+    print("filename = {} ".format(filename))
+    parent = "projects/{}/regions/{}".format(project, region)
+    zone_uri = ("https://www.googleapis.com/compute/v1/projects/{}/zones/{}"
+                .format(project, zone))
+
+    workflow_data = {
+        "placement": {
+            "managed_cluster": {
+                "cluster_name": cluster_name,
+                "config": {
+                    "gce_cluster_config": {"zone_uri": zone_uri},
+                    "master_config": {
+                        "num_instances": 1,
+                        "machine_type_uri": "n1-standard-1",
+                    },
+                    "worker_config": {
+                        "num_instances": 2,
+                        "machine_type_uri": "n1-standard-1",
+                    },
+                },
+            }
+        },
+        "jobs": [
+            {
+                "pyspark_job": {
+                    "main_python_file_uri": "gs://{}/{}".format(
+                        bucket_name, filename)
+                },
+                "step_id": "pyspark-job",
+            }
+        ],
+    }
+
+    workflow = dataproc.instantiate_inline_workflow_template(parent,
+                                                             workflow_data)
+
+    workflow.add_done_callback(callback)
+    global waiting_callback
+    waiting_callback = True
+
+
+def callback(operation_future):
+    # Reset global when callback returns.
+    global waiting_callback
+    waiting_callback = False
+
+
+def wait_for_workflow_end():
+    """Wait for cluster creation."""
+    print("Waiting for workflow completion ...")
+    print('Workflow and job progress, and job driver output available from: '
+          'https://console.cloud.google.com/dataproc/workflows/')
+
+    while True:
+        if not waiting_callback:
+            print("Workflow completed.")
+            break
+
+
+def main(
+    project_id,
+    zone,
+    cluster_name,
+    bucket_name,
+    pyspark_file=None,
+    create_new_cluster=True,
+    global_region=True,
+):
+
+    # [START dataproc_get_client]
+    if global_region:
+        region = "global"
+        # Use the default gRPC global endpoints.
+        dataproc_workflow_client = dataproc_v1.WorkflowTemplateServiceClient()
+    else:
+        region = get_region_from_zone(zone)
+        # Use a regional gRPC endpoint. See:
+        # https://cloud.google.com/dataproc/docs/concepts/regional-endpoints
+        client_transport = (workflow_template_service_grpc_transport
+                            .WorkflowTemplateServiceGrpcTransport(
+                                address="{}-dataproc.googleapis.com:443"
+                                .format(region)))
+        dataproc_workflow_client = dataproc_v1.WorkflowTemplateServiceClient(
+            client_transport
+        )
+    # [END dataproc_get_client]
+
+    try:
+        if pyspark_file:
```

Comment:
I'd prefer you move this logic into `get_pyspark_file` itself.

Reply:
Done in single_job_workflow and submit_job_to_cluster

Reply:
Done or this file and submit_job_to_cluster.